nvflare/private/fed/app/deployer/simulator_deployer.py

```diff
@@ -0,0 +1,164 @@
+# Copyright (c) 2021-2022, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import shutil
+import tempfile
+
+from nvflare.apis.event_type import EventType
+from nvflare.apis.utils.common_utils import get_open_ports
+from nvflare.private.fed.client.admin import FedAdminAgent
+from nvflare.private.fed.client.admin_msg_sender import AdminMessageSender
+from nvflare.private.fed.client.client_req_processors import ClientRequestProcessors
+from nvflare.private.fed.client.fed_client import FederatedClient
+from nvflare.private.fed.simulator.simulator_client_engine import SimulatorClientEngine
+from nvflare.private.fed.simulator.simulator_server import SimulatorServer
+
+from ..server.server_train import create_admin_server
+from .base_client_deployer import BaseClientDeployer
+from .server_deployer import ServerDeployer
+
+
+class SimulatorDeployer(ServerDeployer):
+    def __init__(self, ports=None):
+        super().__init__()
+        if ports:
+            self.open_ports = ports
+        else:
+            self.open_ports = get_open_ports(2)
+        self.admin_storage = tempfile.mkdtemp()
```

Comment:
Should add a prefix here so it's easier to do find the folder and do cleanup.

Reply:
The tempfile.mkdtemp() creates the folder in the standard location, easy for cleanup. Plus the close() will do the cleanup as well.